controlplane/pkg/remote/endpoint_selector_service.go

```diff
@@ -143,16 +138,24 @@ func (cce *endpointSelectorService) validateConnection(ctx context.Context, conn
 		return err
 	}
 
-	result, err := cce.pluginRegistry.GetConnectionPluginManager().ValidateConnection(ctx, conn)
-	if err != nil {
-		return err
+	checkIntersection := func(ip, ipName string) error {
```

Comment:
Better to move it into separate function.

Reply:
- [ ]  move checkIntersection into separate function